### PR TITLE
AG-11450 - Cleanup autoSize option.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-11450

Removes `autoSize` option, as `autoSize: false` was implicit by virtue of setting `width` + `height` options.

Adds `minWidth` and `minHeight` options to replace and allow finer control over the `autoSize: true` case.

Reworks the canvas centering so for better Safari compatibility.

### CODE REVIEW

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/dom/domManager.ts]: What's the `100%` in relation to?

Comment (alantreadway) [reply packages/ag-charts-community/src/chart/dom/domManager.ts]: Discussed offline, we've addressed this by reworking the DOM and moving which elements play which roles.

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/dom/domManager.ts]: This should be equivalent to

```suggestion
        style.width = `${optionsWidth ?? minWidth}px`;
        style.height = `${optionsHeight ?? minHeight}px`;
```

The advantage of explicit sizes are the browser can optimise its layout calculations better

File: packages/ag-charts-community/src/options/chart/chartOptions.ts
Comment (jacobp100) [inline packages/ag-charts-community/src/options/chart/chartOptions.ts:153]: Only thing about this naming is `height` overrides it. Maybe `preferredMinHeight` or something - or just adjust the documentation to mention height is an override

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/dom/domManager.ts:54]: Is `ag-charts-canvas-overlay` still used? What does `display: contents` do here?

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/dom/domManager.ts:54]: Yes it's still in use - it just skips rendering/calculation/contribution of of margin/padding/border from that element, and just displays contents.

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/dom/domManager.ts]: Tested it out and you are right - fixed.

File: packages/ag-charts-community/src/options/chart/chartOptions.ts
Comment (alantreadway) [reply packages/ag-charts-community/src/options/chart/chartOptions.ts:153]: I've adjusted the documentation, thanks!

File: packages/ag-charts-community/src/chart/dom/domManager.ts
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/dom/domManager.ts:54]: Probably need to see it in the browser to follow what's happening. Don't hold off merging, I'll look after it's merged

**Pull request merged**
